nixos/modules/services/misc/transfer-sh.nix

```diff
@@ -0,0 +1,434 @@
+        description = "redirect to https";
+        default = "false";
+        type = types.nullOr types.str;
+      };
+
+      TLS_LISTENER = mkOption {
+        description = "port to use for https";
+        default = null;
+        type = types.nullOr types.int;
+      };
+
+      TLS_LISTENER_ONLY = mkOption {
+        description = "flag to enable tls listener only";
+        default = "false";
+        type = types.nullOr types.str;
+      };
+
+      TLS_CERT_FILE = mkOption {
+        description = "path to tls certificate";
+        default = null;
+        type = types.nullOr types.str;
+      };
+
+      TLS_PRIVATE_KEY = mkOption {
+        description = "path to tls private key";
+        default = null;
+        type = types.nullOr types.str;
+      };
+
+      HTTP_AUTH_USER = mkOption {
+        description = "user for basic http auth on upload";
+        default = null;
+        type = types.nullOr types.str;
+      };
+
+      HTTP_AUTH_PASS = mkOption {
+        description = "pass for basic http auth on upload";
+        default = null;
+        type = types.nullOr types.str;
+      };
+
+      HTTP_AUTH_HTPASSWD = mkOption {
+        description = "htpasswd file path for basic http auth on upload";
+        default = null;
+        type = types.nullOr types.str;
+      };
+
+      HTTP_AUTH_IP_WHITELIST = mkOption {
+        description = "comma separated list of ips allowed to upload without being challenged an http auth";
+        default = null;
+        type = types.nullOr types.str;
+      };
+
+      IP_WHITELIST = mkOption {
+        description = "comma separated list of ips allowed to connect to the service";
+        default = null;
+        type = types.nullOr types.str;
+      };
+
+      IP_BLACKLIST = mkOption {
+        description = "comma separated list of ips not allowed to connect to the service";
+        default = null;
+        type = types.nullOr types.str;
+      };
+
+      TEMP_PATH = mkOption {
+        description = "path to temp folder";
+        default = null;
+        type = types.nullOr types.str;
+      };
+
+      WEB_PATH = mkOption {
+        description = "path to static web files (for development or custom front end)";
+        default = null;
+        type = types.nullOr types.str;
+      };
+
+      PROXY_PATH = mkOption {
+        description = "path prefix when service is run behind a proxy";
+        default = null;
+        type = types.nullOr types.str;
+      };
+
+      PROXY_PORT = mkOption {
+        description = "port of the proxy when the service is run behind a proxy";
+        default = null;
+        type = types.nullOr types.int;
+      };
+
+      EMAIL_CONTACT = mkOption {
+        description = "email contact for the front end";
+        default = null;
+        type = types.nullOr types.str;
+      };
+
+      GA_KEY = mkOption {
+        description = "google analytics key for the front end";
+        default = null;
+        type = types.nullOr types.str;
```

Comment:
I hope no one sets this one

Reply:
Should I take it out?

Reply:
plain text secrets are never fun. If an alternative way to get the secret to the service exists you should certainly be using that, I'd say this should be removed if there is any alternative.